lib/signet/oauth_2/client.rb

```diff
@@ -782,7 +782,13 @@ def expires_at
       # @param [String,Integer,Time] new_expires_at
       #    The access token issuance time.
       def expires_at=(new_expires_at)
-        @expires_at = normalize_timestamp(new_expires_at)
+        new_timestamp = normalize_timestamp new_expires_at
+        if new_timestamp.nil?
+          self.expires_in = nil
+        else
+          self.expires_in = new_timestamp - Time.now
+        end
+        self.expires_at
```

Comment:
I'm pretty sure Ruby ignores the return value of a setter method. And if not, this would be the old value, which you don't want to return.

Reply:
Yeah, good catch.